free-1.md

```diff
@@ -1,68 +1,198 @@
+a related field trying to make a drawing to explain an idea, or perhaps making
+a drawing as a part of figuring out their ideas.
+
+Notably these people are careful and analytical, but they may not be
+computer-savvy.
 
+I want to build a language that allows them to make these drawings in a way
+that saves time and feels natural to them.
 
 ### What's the need?
-_What need is met by your idea? Who are you helping? What is that person's
-experience like now? What would their experience be like if you could help 
-them?_
 
+The need is to create drawings, usually to be embedded in some sort of text
+document, but not always.
+
+Right now people do this in a few ways. The high school teachers I know use
+various plugins for Microsoft Word that have graphical interfaces. Others
+choose to use LaTeX and systems built on `tikz` like `tkz-euclid`, found
+[here](https://www.ctan.org/pkg/tkz-euclide?lang=en).
+
+The graphical interfaces are nice in some situations, but ambiguities emerge
+sometimes, because the only input to the program is (x,y) position, it may not
+always be clear what geometrically significant idea that location corresponds
+to. For example, if I draw a line from one angle of a triangle to the other
+side, am I drawing to the side's midpoint or bisecting the angle? These are
+usually not the same, but are close to each other!
+
+The tkz-euclid solution on the other hand, has a static textual interface. You
+can draw lines, draw shapes, and label intersections. The system is
+impressively complete, but seems cryptic at times. It is not clear how to do
+things like find an intersection, and more generally you don't always know what
+options are available to you. Furthermore, the names of the language correspond
+vaguely to the mathematical writing that they are based on, but are heavily
+constrained by LaTeX
+
+I would like to create an experience that allows for more complete interaction
+between the user and the computer. It would be awesome to allow for users to
+express what they would like to draw in a language and syntax that feels
+natural to them, while also getting visual feedback about what they're drawing
+and what they could choose to do next.
 
 ### Why a language?
-_Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+I think that I should address this question in two parts:
+   1. This question should be solved with a DSL because geometric writing is
+   already designed to unambiguously communicate geometric constructions. There
+   we should try to teach computers how to understand that language, or
+   something similar to it.
+   1. Why interactive? Is making something interactive really part of DSLs? Is
+   making something interactive really changing the language? I argue yes.
+   Language is fundamentally a framework for communication. To me it seems odd
+   that programming languages so far have just been about users telling
```

Comment:
This is an interesting idea.  I'd be very curious to see how this language did it - do you have any examples of programming languages that behave in this fashion?